[HOTFIX] update holonix commands to get recommended version (#497)
holochain/docs-pages: src/pages/get-started/install-advanced.md

The flake-based one-liner to get you an ad-hoc Holonix shell looks like this:

```shell
nix develop github:holochain/holonix?ref=main-0.3
```

> The above one-liner will give you the latest version of the **recommended** Holochain 0.3 release branch. To get an ad-hoc shell with a specific version of Holochain, change the `ref` parameter, for example:

Reviewer: ```suggestion
The above one-liner will give you the latest **recommended** version of Holochain from the 0.3 release branch. To get an ad-hoc shell with a specific version of Holochain, change the `ref` parameter, for example:
```

```shell
nix develop github:holochain/holonix?ref=main-0.4
```

The options you should know about are:

> * `main` or no `ref` parameter: Bleeding-edge development version of Holochain (currently 0.5)

Reviewer: ```suggestion
* `main` or no `ref` parameter: The development version of Holochain, released weekly with no guarantee of stability (currently 0.5)
```

* `main-0.4`: RC of Holochain 0.4 (be aware that many breaking changes are coming in the next RC)
* `main-0.3`: The current recommended version of Holochain for everyday development

> * `main-0.2`: An older version of Holochain, no longer recommended

Reviewer: ```suggestion
```

This version isn't supported for the new Holonix. Existing users at 0.2 should continue to use the old Holonix until they upgrade to 0.3+
